test/abilities/truant.test.ts

```diff
@@ -0,0 +1,55 @@
+import { AbilityId } from "#enums/ability-id";
+import { MoveId } from "#enums/move-id";
+import { MoveResult } from "#enums/move-result";
+import { SpeciesId } from "#enums/species-id";
+import { GameManager } from "#test/test-utils/game-manager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
+
+describe("Ability - Truant", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .ability(AbilityId.BALL_FETCH)
+      .battleStyle("single")
+      .criticalHits(false)
+      .moveset([MoveId.SPLASH])
+      .ability(AbilityId.TRUANT)
+      .enemySpecies(SpeciesId.MAGIKARP)
+      .enemyAbility(AbilityId.BALL_FETCH)
+      .enemyMoveset(MoveId.SPLASH)
+      .startingLevel(100)
+      .enemyLevel(100);
+  });
+
+  it("Truant should block moves ever other turn", async () => {
+    await game.classicMode.startBattle([SpeciesId.FEEBAS]);
+
+    const player = game.field.getPlayerPokemon();
+
+    game.move.select(MoveId.SPLASH);
+    await game.toNextTurn();
+    expect(player.getLastXMoves(1)[0].result).toBe(MoveResult.SUCCESS);
+
+    game.move.select(MoveId.SPLASH);
+    await game.toNextTurn();
+    expect(player.getLastXMoves(1)[0].result).toBe(MoveResult.FAIL);
```

Comment:
```suggestion
    expect(player.getLastXMoves(1)[0].move).toBe(MoveId.NONE);
    expect(karp.hp).toBe(karp.getMaxHp());
    expect(game.textInterceptor.logs).toContain(i18next.t("battlerTags:truantLapse", {
      pokemonNameWithAffix: getPokemonNameWithAffix(player),
    });
```

_Teeechincally_ the move doesn't fail, it just don't do anything.
(This is actually pertinent for Stomping Tantrum)